### ISSUE TITLE
Duplicate axis error

### ISSUE THREAD

**Issue by asarnow** [asarnow/pyem#97]
Caused by an index operation that requires only unique index values in the data. Not sure what's up with it.

```
Writing MovieExport/Movies/stack_07738_X-1Y+1-3_patch_aligned_doseweighted.star
Traceback (most recent call last):
File “/opt/cryoem/pyem/csparc2star.py”, line 170, in
sys.exit(main(parser.parse_args()))
File “/opt/cryoem/pyem/csparc2star.py”, line 71, in main
star.write_star(mic_star, data_general[[f for f in fields if f in data_general]])
File “/opt/cryoem/pyem/pyem/star.py”, line 546, in write_star
df_optics = gb[df.columns.intersection(Relion.OPTICSGROUPTABLE)].first().reset_index(drop=False)
File “/opt/local/miniconda2/envs/pyem/lib/python3.8/site-packages/pandas/core/groupby/groupby.py”, line 1579, in first
return self._agg_general(
File “/opt/local/miniconda2/envs/pyem/lib/python3.8/site-packages/pandas/core/groupby/groupby.py”, line 999, in _agg_general
return self._cython_agg_general(
File “/opt/local/miniconda2/envs/pyem/lib/python3.8/site-packages/pandas/core/groupby/generic.py”, line 1019, in _cython_agg_general
agg_blocks, agg_items = self._cython_agg_blocks(
File “/opt/local/miniconda2/envs/pyem/lib/python3.8/site-packages/pandas/core/groupby/generic.py”, line 1030, in _cython_agg_blocks
data: BlockManager = self._get_data_to_aggregate()
File “/opt/local/miniconda2/envs/pyem/lib/python3.8/site-packages/pandas/core/groupby/generic.py”, line 1695, in _get_data_to_aggregate
obj = self._obj_with_exclusions
File “pandas/_libs/properties.pyx”, line 33, in pandas._libs.properties.CachedProperty.get
File “/opt/local/miniconda2/envs/pyem/lib/python3.8/site-packages/pandas/core/base.py”, line 204, in _obj_with_exclusions
return self.obj.reindex(columns=self._selection_list)
File “/opt/local/miniconda2/envs/pyem/lib/python3.8/site-packages/pandas/util/_decorators.py”, line 309, in wrapper
return func(*args, **kwargs)
File “/opt/local/miniconda2/envs/pyem/lib/python3.8/site-packages/pandas/core/frame.py”, line 4031, in reindex
return super().reindex(**kwargs)
File “/opt/local/miniconda2/envs/pyem/lib/python3.8/site-packages/pandas/core/generic.py”, line 4458, in reindex
return self._reindex_axes(
File “/opt/local/miniconda2/envs/pyem/lib/python3.8/site-packages/pandas/core/frame.py”, line 3871, in _reindex_axes
frame = frame._reindex_columns(
File “/opt/local/miniconda2/envs/pyem/lib/python3.8/site-packages/pandas/core/frame.py”, line 3916, in _reindex_columns
return self._reindex_with_indexers(
File “/opt/local/miniconda2/envs/pyem/lib/python3.8/site-packages/pandas/core/generic.py”, line 4521, in _reindex_with_indexers
new_data = new_data.reindex_indexer(
File “/opt/local/miniconda2/envs/pyem/lib/python3.8/site-packages/pandas/core/internals/managers.py”, line 1276, in reindex_indexer
self.axes[axis]._can_reindex(indexer)
File “/opt/local/miniconda2/envs/pyem/lib/python3.8/site-packages/pandas/core/indexes/base.py”, line 3285, in _can_reindex
raise ValueError(“cannot reindex from a duplicate axis”)
ValueError: cannot reindex from a duplicate axis
```

**Comment by HamishGBrown** [comment]
Using 9c16200 from master branch. I worked around this issue by adding the following line:

`data_general = data_general.loc[:,~data_general.columns.duplicated()]`

before 71 in csparc2star.py

**Comment by asarnow** [comment]
Fixed by 4d980ee6.

**Comment by asarnow** [comment]
PS Thank you @HamishGBrown. I went with
`df.drop(df.columns[df.columns.duplicated()], axis=1, inplace=True)`
instead, I believe it has a smaller memory footprint.

It'll also be done when writing any .star file, not just the movie ones.
